torch/_prims/rng_prims.py

```diff
@@ -0,0 +1,51 @@
+from typing import Tuple
+
+import torch
+from torch import _prims
+from torch._decomp.decompositions_for_rng import RNGStateHelper, throw_on_cpu
+from torch.types import _device, _dtype
+
+
+def _philox_rand(
+    shape: torch.Size,
+    seed: torch.Tensor,
+    offset: torch.Tensor,
+    stride: Tuple[int, ...],
+    device: _device,
+    dtype: _dtype,
+):
+    # stride arg will be useful for distributed usecase. Currently, its unused.
+    if device.type == "cpu":
+        devices = []
+    else:
+        devices = [device]
+
+    if device.type == "cpu":
+        raise throw_on_cpu()
+    devices = [device]
+
+    with torch.random.fork_rng(devices):
```

Comment:
This only works for cpu and cuda right now (there is a PR to add more).
So you should raise an eeror for non-supported device.